### PR TITLE
[#1853] another attempt to fix Storage2UriMapperJavaImpl race

### PR DESCRIPTION
**Pull request by cdietrich**
[#1853] another attempt to fix Storage2UriMapperJavaImpl race

### CODE REVIEW

File: org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/resource/Storage2UriMapperJavaImpl.java [hunk 1]
```diff
@@ -368,7 +368,7 @@ public Iterable<Pair<IStorage, IProject>> getStorages(/* @NonNull */ URI uri) {
 		// Pessimistic copy of the cachedPackageFragmentRootData
 		Collection<PackageFragmentRootData> packageFragmentRootDatas;
 		synchronized(cachedPackageFragmentRootData) {
-			packageFragmentRootDatas = cachedPackageFragmentRootData.values();
+			packageFragmentRootDatas = newArrayList(cachedPackageFragmentRootData.values());
```
Comment (cdietrich) [inline org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/resource/Storage2UriMapperJavaImpl.java:371]: @szarnekow do you know why this was not there in the first place

Comment (cdietrich) [reply org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/resource/Storage2UriMapperJavaImpl.java:371]: https://github.com/eclipse/xtext-eclipse/issues/1853#issuecomment-1174731216

Comment (szarnekow) [reply org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/resource/Storage2UriMapperJavaImpl.java:371]: A copy of the list on every read is probably not what we want to do.
Maybe use a synchronized block on the list itself after it was read from the map?

Comment (cdietrich) [reply org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/resource/Storage2UriMapperJavaImpl.java:371]: hmmmm ok. will try it out as workaround in our codebase

Comment (cdietrich) [reply org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/resource/Storage2UriMapperJavaImpl.java:371]: we still need to check if the problem happened again

**Comment by cdietrich** [comment]
as i wont find time to further work on this => close

**Pull request closed without merging**